harness/determined/common/experimental/experiment.py

```diff
@@ -98,6 +98,24 @@ def delete(self) -> None:
     def get_config(self) -> Dict[str, Any]:
         return self._get().config
 
+    def get_trials(
+        self,
+        sort_by: trial.TrialSortBy = trial.TrialSortBy.ID,
+        order_by: trial.TrialOrderBy = trial.TrialOrderBy.ASCENDING
+    ) -> List[trial.TrialReference]:
+        """
+        Get the list of :class:`~determined.experimental.TrialReference` instances
+        representing trials for an experiment.
+
+        Arguments:
+            sort_by: Which field to sort by. See :class:`~determined.experimental.TrialSortBy`.
+            order_by: Whether to sort in ascending or descending order. See
+                :class:`~determined.experimental.TrialOrderBy`.
+        """
+        r = self._session.get(f"/api/v1/experiments/{self.id}/trials")
+        trials = r.json().get("trials")
```

Comment:
maybe hard fail here if `trials` field is missing (probably meaning something is very wrong)?

```suggestion
        trials = r.json()["trials"]
```

Reply:
Good thing you asked; I forgot to mention it in my comment. I was following what I though was a pattern established in
https://github.com/determined-ai/determined/blob/master/harness/determined/common/experimental/determined.py#L156
https://github.com/determined-ai/determined/blob/master/harness/determined/common/experimental/determined.py#L165
https://github.com/determined-ai/determined/blob/master/harness/determined/common/experimental/determined.py#L196
https://github.com/determined-ai/determined/blob/master/harness/determined/common/experimental/determined.py#L205
(In all of the above examples, `None` is not a viable result; another exception is imminent if `get` returns `None`).
However, upon closer look, we also have it the other way in a few instances:
https://github.com/determined-ai/determined/blob/master/harness/determined/common/experimental/experiment.py#L205
and 6 or 7 more.

Should we homogenize the code? (There is, of course, a risk of regressions).

Reply:
I suspect that in 90% of these cases if `.get(` doesn't have `default=` parameter than it's probably "useless". Master normally returns these fields so we never run into a situation where it'd blow up.

For example in https://github.com/determined-ai/determined/blob/master/harness/determined/common/experimental/determined.py#L156
it won't fail immediately, but then it'll fail in `from_json` on this lookup: https://github.com/determined-ai/determined/blob/9e91c01bdb8f591f4978654f1d7b1b15677156d3/harness/determined/common/experimental/model.py#L351

In https://github.com/determined-ai/determined/blob/master/harness/determined/common/experimental/determined.py#L205 it'd make more sense to do `.get("labels", [])`, otherwise it's also not looking right.